core/server/apps/subscribers/lib/helpers/input_email.js

```diff
@@ -20,6 +20,10 @@ module.exports = function input_email(options) { // eslint-disable-line camelcas
         extras += 'autofocus="autofocus"';
     }
 
+    if (options.hash.required) {
```

Comment:
To avoid copying the `subscribe_form.hbs` into the theme, we could allow the option

`email_required="true"` 

for the `subscribe_form` helper.